src/components/Auth/Login.jsx

```diff
@@ -0,0 +1,184 @@
+import "./auth-components.css";
+import { useEffect, useState } from "react";
+import { Link, Navigate, useLocation, useNavigate } from "react-router-dom";
+import { useAuthFormData, useDocumentTitle } from "../../custom-hooks";
+import { useAuth } from "../../contexts";
+import { initiateLogin, setAuthStateInLocalStorage } from "../../utils";
+import { LoadingAnimation } from "../../components";
+
+const Login = ({ activeAuthComponentDisplay }) => {
+  const { authState, setAuthState } = useAuth();
+  const { isAuthenticated, authError } = authState;
+  const [authFormDataState, authFormDataDispatch] = useAuthFormData();
+  const { email, password, rememberMe } = authFormDataState;
+  const [showPassword, setShowPassword] = useState();
+  const [loadingState, setLoadingState] = useState(false);
```

Comment:
You can move the loading state to the auth state in its context. No need of this IG.

Reply:
Yes, I think it can be done, can even lift the state more higher because it's needed at many places! Will do that.